File: pails/pails.cpp
#include <iostream>
#include <fstream>
#include <vector>
using namespace std;

vector<int> v(1000,0);

int solve(int* x,int* y,int* z,int curr){
    if(v[curr] != 0){
        return v[curr];
    }
    if(curr + *x > *z){
        v[curr] = curr;
        return curr;
    }
    else if(curr + *x == *z){
        v[curr] = curr + *x;
        return curr + *x;
    }
    if(curr + *y < *z){
        v[curr] = max(solve(x,y,z,curr + *y),solve(x,y,z,curr+*x));
        return v[curr];
    }
    v[curr] = solve(x,y,z,curr+*x);
    return v[curr];
}


int main(){
    ifstream of;
    of.open("pails.in");
    ofstream write;
    write.open("pails.out");
    int x,y,z;
    of>>x>>y>>z;

    write <<   solve(&x,&y,&z,0);
    write.close();
    of.close();
}

File: pails/pails.py
from typing import List


f = open("pails.in","r")
w = open("pails.out","w")


x = [False] * 1000
x,y,z = map(int,f.readline().strip().split())



def writeToOut(curr):
    global out
    global outlen
    if curr > out:
        out = curr
def solve(curr,lens):
    if x[curr]: return
    if curr+x>z:
        x[curr] = True
        writeToOut(curr)
        return
    elif curr+x == z:
        writeToOut(curr+x)
    if curr+y<=z:
        solve(curr+y,lens+1)
    solve(curr+x,lens+1)
        
solve(0,0)
print(out,file=w)
f.close()
w.close()